### DISCUSSION TITLE
Hosted Bitbucket Cloud App

### DISCUSSION THREAD

**Discussion by setchy** [renovatebot/renovate#14700] [Suggest an Idea]
I'm interested in what feature enhancements would be needed in order to provide a hosted Bitbucket Cloud App, similar to the awesome [hosted GitHub App](https://docs.renovatebot.com/getting-started/installing-onboarding/#hosted-githubcom-app) or [hosted GitLab.com App](https://docs.renovatebot.com/getting-started/installing-onboarding/#hosted-gitlabcom-app)

From searching through open issues and this discussion boards, it seems a common prerequisite is the availability of a platform/workspace level webhook, such that Renovate can listen to VCS events, queue up request and process each in-turn.

For Bitbucket Cloud, they do support [Create a webhook for a workspace](https://developer.atlassian.com/cloud/bitbucket/rest/api-group-workspaces/#api-workspaces-workspace-hooks-post) via their REST API which on the surface appears to be suitable.  I don't believe Atlassian provide this via their Administrative UI

Alternatively, perhaps adding support for Bitbucket Cloud in [whitesource/renovate-on-prem](https://github.com/whitesource/renovate-on-prem/issues/148) would be a simpler path forward 🤔

**Comment by rarkins** [comment]
I'm interested in this too, so any further research you can do would help me move it forward.

Highest priority is to work out what auth/identity model to use. For example, if we have @renovate-bot on Bitbucket Cloud and projects invite it then we likely would hit rate limits pretty fast. Is there any modern "app" approach similar to GitHub's which is available? If no then we're probably left with the idea of OAuth. This means that PRs would come from yourself instead of a "bot", which is undesirable, but still better than nothing.

Webhooks are desirable although not 100% essential. Without them we can't do interactive things, but at least we could run on a schedule. This part is a little dependent on the first part about identity model.

**Comment by setchy** [comment]
Atlassian Forge is the latest framework for creating Bitbucket Cloud apps/extensions - https://developer.atlassian.com/platform/forge/, but not clear if it supports Bitbucket from the documentation 

The other option is the Atlassian Connect framework - https://developer.atlassian.com/cloud/bitbucket/integrating-with-bitbucket-cloud/#what-is-atlassian-connect-

**Reply by setchy** [reply]
Ok, so Atlassian Forge doesn't have an explicit Bitbucket Cloud module/template, but it does have

- [Scheduled trigger](https://developer.atlassian.com/platform/forge/manifest-reference/modules/scheduled-trigger/)
- [Trigger](https://developer.atlassian.com/platform/forge/manifest-reference/modules/trigger/)
- [Web trigger](https://developer.atlassian.com/platform/forge/manifest-reference/modules/web-trigger/)

**Reply by rarkins** [reply]
Do any of these support "app" semantics similar to GitHub? Eg install the app into an org or select repos, webhooks get set up automatically, etc?

**Reply by setchy** [reply]
I believe so.

**Reply by setchy** [reply]
Forge for Bitbucket is coming soon, according to this published [q1 2022 roadmap](https://blog.developer.atlassian.com/webinar-recap-forge-roadmap-q1-2022/)
